metricbeat/helper/prometheus/metric.go

```diff
@@ -37,6 +38,25 @@ type MetricMap interface {
 
 	// GetValue returns the resulting value
 	GetValue(m *dto.Metric) interface{}
+
+	// GetConfiguration returns the configuration for the metric
+	GetConfiguration() Configuration
+}
+
+// Configuration for mappings that needs extended treatment
+type Configuration struct {
+	// StoreNonMappedLables indicates if labels found at the metric that are
+	// not found at the label map should be part of the resulting event
+	StoreNonMappedLabels bool
+	// NonMappedLabelsPlacement is used when StoreNonMappedLabels is set to true, and
+	// defines the key at the event to store labels
+	NonMappedLabelsPlacement string
```

Comment:
This means that we by-pass the mapping https://github.com/elastic/beats/pull/14794/files#diff-dec796600491bdd3547c0d5777fb1ea6R31?

Reply:
no, this means that if `StoreNonMappedLabels` is set to true, and there is a metric family that contains labels whose names are not known beforehand but we are interested in pushing with our events, they will be added, and placed under the location of `NonMappedLabelsPlacement`

By your comment it looks like this is not clear and that I should rethink.

Would a comment like:
```
// NonMappedLabelsPlacement is used when StoreNonMappedLabels is set to true, and
// defines the key path at the event under which to store the dynamically found labels. This key path
// will be added to the events that match this metric and a subset of key, value pairs will be created under
// it, one for each non mapped label found
```

please, propose/modify which whatever you find would be clarifying

Reply:
Yeap, got it now! I think it will be helpful if we add an example here showing the expected output. I see nice "examples" in the unit-tests, how about adding one in this docstring?

Reply:
@ChrsMark updated